src/solvers/smt/smt_conv.cpp

```diff
@@ -2241,46 +2252,41 @@ const struct_union_data &smt_convt::get_type_def(const type2tc &type) const
            : dynamic_cast<const struct_union_data &>(*type.get());
 }
 
-smt_astt smt_convt::array_create(const expr2tc &expr)
+smt_astt smt_convt::array_create(
```

Comment:
Both vector and array share the same base class, so you can change the method to accept array_data, instead of passing redudant information in the argument (e.g., the `is_infinite`, `size` and `members` are already part of `expr`).

Reply:
Is this still valid?